pkg/parser/parser.go

```diff
@@ -1,42 +1,121 @@
 package parser
 
 import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"path/filepath"
+
 	"github.com/instrumenta/conftest/pkg/parser/cue"
 	"github.com/instrumenta/conftest/pkg/parser/ini"
 	"github.com/instrumenta/conftest/pkg/parser/terraform"
 	"github.com/instrumenta/conftest/pkg/parser/toml"
 	"github.com/instrumenta/conftest/pkg/parser/yaml"
 )
 
+// ValidInputs returns string array in order to passing valid input types to viper
+func ValidInputs() []string {
+	return []string{
+		"toml",
+		"tf|hcl",
+		"cue",
+		"ini",
+		"yaml",
+	}
+}
+
 // Parser is the interface implemented by objects that can unmarshal
 // bytes into a golang interface
 type Parser interface {
 	Unmarshal(p []byte, v interface{}) error
 }
 
-// GetParser returns a Parser for the given input type. Defaults to returning the YAML parser.
-func GetParser(i *Input) Parser {
+// ConfigDoc stores file contents and it's original filename
+type ConfigDoc struct {
+	ReadCloser io.ReadCloser
+	Filepath   string
+}
+
+// ReadUnmarshaller is an interface that allows for bulk unmarshalling
+// and setting of io.Readers to be unmarshalled.
+type ReadUnmarshaller interface {
+	BulkUnmarshal(readerList []ConfigDoc) (map[string]interface{}, error)
+}
+
+// ConfigManager the implementation of ReadUnmarshaller and io.Reader
+// byte storage.
+type ConfigManager struct {
+	parser         Parser
+	configContents map[string][]byte
+}
+
+// BulkUnmarshal iterates through the given cached io.Readers and
+// runs the requested parser on the data.
+func (s *ConfigManager) BulkUnmarshal(configList []ConfigDoc) (map[string]interface{}, error) {
+	err := s.setConfigs(configList)
+	if err != nil {
+		return nil, fmt.Errorf("we should not have any errors on setting our readers: %v", err)
```

Comment:
Messages elsewhere are leading caps. Also, this message (ie. starting with "we") feels like a message to the developers, not the end user. I would suggest rewriting around the user problem.

Reply:
Reframed these messages to reference the user, and capitalized.